pylabrobot/centrifuge/agilent_centrifuge.py

```diff
@@ -0,0 +1,787 @@
+  "aa 02 0e 10",
+  "aa 01 0e 0f",
+  "aa 02 0e 10",
+  "aa 02 0e 10",
+  "aa 01 0e 0f",
+  "aa 02 0e 10",
+  "aa 01 17 02 1a",
+  "aa 01 0e 0f",
+  "aa 01 e6 c8 00 b0 04 96 00 0f 00 4b 00 a0 0f 05 00 07",
+  "aa 01 17 04 1c",
+  "aa 01 17 01 19",
+  "aa 01 0e 0f",
+  "aa 02 0e 10",
+  "aa 01 0b 0c",
+  "aa 01 0e 0f",
+  "aa 02 0e 10",
+  "aa 01 17 02 1a",
+  "aa 01 0e 0f",
+  "aa 01 e6 c8 00 b0 04 96 00 0f 00 4b 00 a0 0f 05 00 07",
+  "aa 01 17 04 1c",
+  "aa 01 17 01 19",
+  "aa 01 0e 0f",
+  "aa 02 0e 10",
+  "aa 01 0b 0c",
+  "aa 01 0e 0f",
+  "aa 01 e6 c8 00 b0 04 96 00 0f 00 4b 00 a0 0f 05 00 07",
+  byte_string,
+  "aa 01 0e 0f",
+  "aa 01 0e 0f",
+  "aa 01 0e 0f",
+  "aa 02 0e 10",
+  "aa 01 0e 0f",
+  "aa 01 0e 0f",
+  "aa 02 0e 10",
+  "aa 01 0e 0f",
+  "aa 01 0e 0f",
+  "aa 02 0e 10",
+  "aa 01 0e 0f",
+  "aa 01 0e 0f",
+  "aa 01 0e 0f",
+  "aa 02 0e 10",
+  "aa 01 0e 0f",
+  "aa 01 0e 0f",
+  "aa 02 0e 10",
+  "aa 01 0e 0f",
+  "aa 01 0e 0f",
+  "aa 01 0e 0f",
+  "aa 02 0e 10",
+  "aa 01 0e 0f",
+  "aa 01 0e 0f",
+  "aa 02 0e 10",
+  "aa 01 0e 0f",
+  "aa 01 0e 0f",
+  "aa 01 0e 0f",
+  "aa 01 0e 0f",
+  "aa 02 0e 10",
+  "aa 01 0e 0f",
+  "aa 02 0e 10",
+  "aa 01 0e 0f",
+  "aa 02 0e 10",
+  "aa 01 0e 0f",
+  "aa 02 0e 10",
+  "aa 01 0e 0f",
+  "aa 02 0e 10",
+  "aa 01 0e 0f",
+  "aa 02 0e 10",
+  "aa 01 0e 0f",
+  "aa 02 0e 10",
+  "aa 01 0e 0f",
+  "aa 02 0e 10",
+  "aa 01 17 02 1a",
+  "aa 01 0e 0f",
+  "aa 02 0e 10",
+  "aa 02 0e 10",
+  "aa 02 26 00 01 29",
+  "aa 02 0e 10",
+  "aa 01 0e 0f",
+  "aa 02 0e 10",
+  "aa 02 0e 10",
+  "aa 02 0e 10",
+  "aa 01 0e 0f",
+  "aa 02 0e 10",
+  "aa 02 0e 10",
+  "aa 01 0e 0f",
+  "aa 02 0e 10",
+  "aa 02 0e 10",
+  "aa 01 0e 0f",
+  "aa 02 0e 10",
+  "aa 02 0e 10",
+  "aa 02 0e 10",
+  "aa 01 0e 0f",
+  "aa 02 0e 10",
+  "aa 02 0e 10",
+  "aa 02 0e 10",
+  "aa 02 26 00 05 2d",
+  "aa 02 0e 10",
+  "aa 01 0e 0f",
+  "aa 02 0e 10",
+  "aa 02 0e 10",
+  "aa 01 0e 0f",
+  "aa 02 0e 10",
+  "aa 01 0e 0f",
+  "aa 02 0e 10",
+  "aa 02 0e 10",
+  "aa 02 26 00 07 2f",
+  "aa 02 0e 10",
+  "aa 01 0e 0f",
+  "aa 02 0e 10",
+  "aa 02 0e 10",
+  "aa 02 0e 10",
+  "aa 01 0e 0f",
+  "aa 02 0e 10",
+  "aa 02 0e 10",
+  "aa 01 0e 0f",
+  "aa 02 0e 10",
+  "aa 02 0e 10",
+  "aa 01 0e 0f",
+  "aa 02 0e 10",
+  "aa 02 0e 10",
+  "aa 02 0e 10",
+  "aa 01 0e 0f",
+  "aa 02 0e 10",
+  "aa 02 0e 10",
+  "aa 02 0e 10",
+  "aa 01 0e 0f",
+  "aa 01 0e 0f",
+  "aa 02 0e 10",
+  "aa 01 0e 0f",
+  "aa 02 0e 10",
+  "aa 01 0e 0f",
+    ]
+
+    for tx in tx_payloads:
+      if isinstance(tx, str):
+        byte_literal = bytes.fromhex(tx)
+        await self.send(byte_literal)
+      else:
+        await self.send(tx)
+
+    self.current_bucket = 2
+
+  async def start_spin_cycle(
+  self,
+  g: Optional[float] = 500,
+  time_seconds: Optional[float] = 10,
+  acceleration: Optional[float] = 80,
+ ) -> None:
+    """Start a spin cycle."""
+    if acceleration < 0:
+      acceleration = acceleration*(-1)
```

Comment:
does this mean opposite direction?

Reply:
This is just to handle if a user puts a negative number for whatever reason. do you think its necessary?

Reply:
i think it's better to have the user specify things explicitly where possible